src/UserHandler.cpp

```diff
@@ -0,0 +1,241 @@
+        std::cout << "Exited club" + genre << std::endl;
+    }
+    else if (ansCmd == ERROR) {
+        std::cout << ans.getBody() << std::endl;
+    }
+}
+
+void UserHandler::addBook(std::string& line) {
+    std::vector <std::string> cmdParams;
+    split(line, cmdParams, CMD_DELIMITER);
+    std::string genre = cmdParams[1];
+    std::string book = cmdParams[2];
+
+    Frame frame = Frame(_actionFrameCommandMap[ADD_BOOK]);
+    frame.addHeader("destination", genre);
+    frame.setBody(_inventory.getUsername() + "has added the book" + book);
+    sendFrame(frame);
+    _inventory.addBook(genre, book);
+}
+
+void UserHandler::borrowBook(std::string& line) {
+    std::vector <std::string> cmdParams;
+    split(line, cmdParams, CMD_DELIMITER);
+    std::string genre = cmdParams[1];
+    std::string book = cmdParams[2];
+
+    Frame frame = Frame(_actionFrameCommandMap[BORROW_BOOK]);
+    frame.addHeader("destination", genre);
+    frame.setBody(_inventory.getUsername() + "wish to borrow" + book);
+    sendFrame(frame);
+}
+
+void UserHandler::returnBook(std::string& line) {
+    std::vector <std::string> cmdParams;
+    split(line, cmdParams, CMD_DELIMITER);
+    std::string genre = cmdParams[1];
+    std::string book = cmdParams[2];
+
+    Frame frame = Frame(_actionFrameCommandMap[RETURN_BOOK]);
+    frame.addHeader("destination", genre);
+    frame.setBody("Returning" + book + "to" + _inventory.returnBook(book));
+    sendFrame(frame);
+}
+
+void UserHandler::bookStatus(std::string& line) {
+    std::vector <std::string> cmdParams;
+    split(line, cmdParams, CMD_DELIMITER);
+    std::string genre = cmdParams[1];
+
+    Frame frame = Frame(_actionFrameCommandMap[GENRE_BOOK_STATUS]);
+    frame.addHeader("destination", genre);
+    frame.setBody("book status");
+    sendFrame(frame);
+}
+
+void UserHandler::logout(std::string& line) {
+    Frame frame = Frame(_actionFrameCommandMap[LOGOUT]);
+    frame.addHeader("receipt", std::to_string(nextId));
+    sendFrame(frame);
+
+    Frame ans = receiveFrame();
+    FrameCommand ansCmd = ans.getCommand();
+    if (ansCmd == ERROR) {
+        std::cout << ans.getBody() << std::endl;
+    }
+}
+
+void UserHandler::parseLine(std::string line) {
+    UserActionsEnum cmd = findCmd(line);
+    switch (cmd) {
+        case LOGIN: {
+            login(line);
+            break;
+        }
+        case JOIN_GENRE: {
+            joinGenre(line);
+            break;
+        }
+        case EXIT_GENRE: {
+            exitGenre(line);
+            break;
+        }
+        case ADD_BOOK: {
+            addBook(line);
+            break;
+        }
+        case BORROW_BOOK: {
+            borrowBook(line);
+            break;
+        }
+        case RETURN_BOOK: {
+            returnBook(line);
+            break;
+        }
+        case GENRE_BOOK_STATUS: {
+            bookStatus(line);
+            break;
+        }
+        default: // Logout
+            logout(line);
+    }
+}
```

Comment:
?

Reply:
because I deal with all the cases, no point on writing case LOGOUT:...
because than a warning will arise